### PR TITLE
Adapt SkyModel to evaluate on TimeMapAxis

### PR DESCRIPTION
**Pull request by AtreyeeS**
Addresses remaining part of  #4480 - adapt SkyModel.evaluate_geom() and SkyModel.integrate_geom() to work with a TimeMapAxis

### CODE REVIEW

**Comment by AtreyeeS** [comment]
I have adapted `SkyModel.evaluate_geom` to support a TimeMapAxis.
Support for `integrate_geom` needs #4859 to be addressed

This has the loop over axis bins, which is not nice. Any better suggestion will be much appreciated.

The nice thing is this can allow `FluxPointsDataset` to work with n-dim geom. For eg, fitting light curves need not go through the splitting into multiple datasets for each time bin.

Maybe we can merge this now, and I can adapt the `FluxPointsDataset` after this?
The `integrate_geom` functionality is more necessary for `MapEvaluator`, which is not planned to be supported at the moment

**Comment by registerrier** [comment]
Hi @AtreyeeS , #4859 has been merged. Could you rebase?

**Comment by AtreyeeS** [comment]
After rebasing, I adapt `SkyModel.integrate_geom()` as well.
https://github.com/gammapy/gammapy/issues/4480 can be closed once this PR is merged.

**Review by QRemy — COMMENTED** [review]
Thanks @AtreyeeS, looks good, I just pushed a commit to reduce code duplication and left one question :

File: gammapy/modeling/models/cube.py [hunk 1]
```diff
@@ -348,19 +348,48 @@ def evaluate_geom(self, geom, gti=None):
...
+                    integral = self.temporal_model.integral(t1, t2)
+                    temp_eval.append(np.sum(integral))
+                value = (value.T * temp_eval).T
+
+            else:
+                if gti is not None:
+                    integral = self.temporal_model.integral(
+                        gti.time_start, gti.time_stop
+                    )
+                    value = value * np.sum(integral)
```
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply gammapy/modeling/models/cube.py]: `GTI` is needed only to know the time subset of the times in the time bin during which the source was observed. If `GTI is None`, then the temporal model will be evaluated in the full time bin.

**Review by Astro-Kirsty — APPROVED** [review]
Thanks @AtreyeeS

**Review by Astro-Kirsty — APPROVED** [review]

**Review by QRemy — APPROVED** [review]

**Pull request merged**